### DISCUSSION TITLE
Unstable Fraunhofer Maven Repository

### DISCUSSION THREAD

**Discussion by marcgs** [eclipse-edc/Connector#1431] [General]
We have observed the following failure in roughly 1% of the verify workflow. Access to the Fraunhofer Maven repo seems to be unstable.

```
Execution failed for task ':extensions:azure:vault:compileTestJava'.
 > Could not resolve all files for configuration ':extensions:azure:vault:testCompileClasspath'.
    > Could not resolve com.microsoft.azure:azure-client-runtime:[1.1.0,2.0.0).
      Required by:
          project :extensions:azure:vault > com.microsoft.azure:azure-mgmt-resources:1.3.0
       > Failed to list versions for com.microsoft.azure:azure-client-runtime.
          > Unable to load Maven meta-data from https://maven.iais.fraunhofer.de/artifactory/eis-ids-public/com/microsoft/azure/azure-client-runtime/maven-metadata.xml.
             > Could not GET 'https://maven.iais.fraunhofer.de/artifactory/eis-ids-public/com/microsoft/azure/azure-client-runtime/maven-metadata.xml'.
                > Connect to maven.iais.fraunhofer.de:443 [maven.iais.fraunhofer.de/193.175.162.89] failed: connect timed out
```

**Comment by algattik** [comment]
It seems reasonable to move the Fraunhofer repo declaration to the relevant module. This will also reduce the load on it.

**Comment by paullatzelsperger** [comment]
Ultimately the Fraunhofer code should be published on MavenCentral IMO.

**Comment by marcgs** [comment]
@algattik the repo declaration would be required in all `:data-protocols:ids` modules plus all other modules depending from it including all launchers and samples. This ends up with quite a lot of duplicated definitions for this repo. IMO we should try to publish these artifacts to MavenCentral. @ronjaquensel WDYT?

**Comment by juliapampus** [comment]
@marcgs I would support this idea, but we should ask the responsibles why this has not already been done and who can do this asap. Or if, as this repos are now in our orga, we are responsible for this next steps. @mspiekermann?

**Comment by mspiekermann** [comment]
We are still in the process to transfer the former Fraunhofer IAIS repositories of IDS-Serializer and IDS-CodeGeneration from FhG-internal structures to the Eclipse Foundation including all IP-transfer processes (approved for IDS-Serializer already). Within this process we will, as a future step, publish these artifacts to MavenCentral. Internal alignment and planning are ongoing and we will do this asap but may need some more days.

**Comment by marcgs** [comment]
Thanks for the info @mspiekermann . I have assigned this bug to you, please feel free to forward to the person responsible for the migration.

**Comment by juliapampus** [comment]
I guess this is still worth a discussion @mspiekermann @timwirtz86

**Comment by timwirtz86** [comment]
Hi @marcgs, just saw the issue. Regarding your command I have two questions:

1. What are you request from our repository. I am not an expert on maven but to me it seem that you are trying to download  the metadata from a package ending with microsoft/azure/azure-client-runtime ('https://maven.iais.fraunhofer.de/artifactory/eis-ids-public/com/microsoft/azure/azure-client-runtime/maven-metadata.xml') from our repository. This I am not aware of an it is not contained in our artifactory. Can you give me more details ? 
2. We are planing to move the lates artifacts to the central maven repository. What artifacts with which version are relevant to you ?

**Reply by marcgs** [reply]
Hi @timwirtz86 ,

1. The Fraunhofer maven repository is listed in EDC's [Gradle build](https://github.com/agera-edc/DataSpaceConnector/blob/main/build.gradle.kts#L70) as a repository to search for dependencies. Sometimes the build process will try to retrieve artefacts from the Fraunhofer maven repo that are not available. This is an expected situation, after which the build process will try other repositories. The problem is that the Fraunhofer maven repository seemed to be unstable from time to time and not able to handle all requests gracefully (hence the "connect time out" error). I am not able to confirm if this situation persists as of today.
2. EDC depends on "de.fraunhofer.iais.eis.ids.infomodel:java:4.2.7"

**Reply by timwirtz86** [reply]
Thanks @marcgs for clarification. 

Regarding (1.) we will move the latest JARs (5.0.0) of our IDS sources to the central maven repository anyway in the next sprint. We will do so for the java classes of the infomodel 4.2.7. 

Regarding (2.) you might want to discuss whether you want to migrate to infomodel 5.0.0. you find the related Changelog here https://github.com/International-Data-Spaces-Association/Java-Representation-of-IDS-Information-Model/blob/main/Changelog.md . 

I will let you know when the artifacts are available in the central maven repository.

**Comment by richardtreier** [comment]
The aforementioned 1% error rate has been a 30% or higher for us in the past three days. 

We have three different EDC Connector Images that we build in our open source [sovity/edc-extensions](https://github.com/sovity/edc-extensions/) repository, which causes this error rate to really feel more like 60%.

It has been failing our pipelines causing us to restart the pipelines up to 4 times just to get a single PR through, with afterwards the main branch also failing to build.

Do you know who would be a good contact person at the Fraunhofer IAIS regarding this nexus? @mspiekermann

**Reply by mspiekermann** [reply]
@timwirtz86 is the right contact and is already involved in this discussion.

**Comment by jimmarino** [comment]
FYI, this dependency will be eliminated in the upcoming milestone, so moving forward we won't have these issues.

**Comment by kbData** [comment]
we still experience this issue. it there any update?
![image](https://github.com/eclipse-edc/Connector/assets/7373683/8b0edff0-8fdd-4425-bf18-8c64944af306)

**Reply by jimmarino** [reply]
This dependency was removed a long time ago, please update your EDC or build.

Also, as explained in the previous messages, this is not an issue with EDC.
